src/Kriti/CustomFunctions.hs

```diff
@@ -134,12 +135,14 @@ removeNullsF = parserToFunc $ J.withArray "Array" \a -> do
 --
 -- `[[1,2],[3,4]]` -> `[1,2,3,4]`
 -- `["hello", " ", "world"]` -> `["hello world"]`
+-- `[{"a":1, "b":2}, {"b":3, "c":4}]` -> `{"a":1, "b":3, "c":4}`
 concatF :: KritiFunc
 concatF = parserToFunc $ J.withArray "Array" \as -> do
   let l = V.toList as
-      a = J.Array . V.concat <$> traverse (J.withArray "Nested Array" pure) l
-      s = J.String . T.concat <$> traverse (J.withText "Nested String" pure) l
-  a <|> s
+      a = J.Array . fold <$> traverse (J.withArray "Nested Array" pure) l
+      s = J.String . fold <$> traverse (J.withText "Nested String" pure) l
+      o = J.Object . fold . reverse <$> traverse (J.withObject "Nested Object" pure) l
```

Comment:
You could probably use `foldM` instead of `traverse` here to avoid the `fold . reverse`.

Reply:
I think foldM takes a folding function so I'd prefer to use a version without that requirement if possible.

Reply:
The type is actually pretty gnarly:

```haskell
foo :: (Monoid (t1 b), Traversable t1, Applicative t2, Foldable t2) => (a -> t2 b) -> t1 a -> t1 b
foo f = fold . traverse f
```

Plus this doesn't handle the 'reverse' so I'd need to use that fold reversal newtype.

Reply:
Maybe I'm missing something that would make this work though.

Reply:
Sorry `foldM` was wrong. I think you could do something like:
```
bar :: [J.Value] -> J.Parser J.Value
bar l = foldl' (\p val -> fmap J.Object (J.withObject "Nested Object" pure val) <> p) mempty l
```
And then you don't need to traverse the structure multiple times to build the parsers, reverse them, then fold them together.